pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -0,0 +1,296 @@
+	if err != nil {
+		return err
+	}
+	for _, configMap := range configMaps.Items {
+		if strings.HasPrefix(configMap.Name, statusConfigMapName) {
+			if revision, ok := configMap.Data["revision"]; ok {
+				revisionID, err := strconv.Atoi(revision)
+				if err != nil {
+					return err
+				}
+				switch configMap.Data["phase"] {
+				case string(corev1.PodSucceeded):
+					succeededRevisionIDs = append(succeededRevisionIDs, revisionID)
+				case string(corev1.PodFailed):
+					failedRevisionIDs = append(failedRevisionIDs, revisionID)
+				}
+			}
+		}
+	}
+
+	sort.Ints(succeededRevisionIDs)
+	sort.Ints(failedRevisionIDs)
+
+	// Get list of protected IDs
+	protectedSucceededRevisionIDs := succeededRevisionIDs[revisionKeyToStart(len(succeededRevisionIDs), c.succeededRevisionLimit):]
+	protectedFailedRevisionIDs := failedRevisionIDs[revisionKeyToStart(len(failedRevisionIDs), c.failedRevisionLimit):]
+
+	excludedIDs := make([]int, 0, len(protectedSucceededRevisionIDs)+len(protectedFailedRevisionIDs))
+	excludedIDs = append(excludedIDs, protectedSucceededRevisionIDs...)
+	excludedIDs = append(excludedIDs, protectedFailedRevisionIDs...)
+	sort.Ints(excludedIDs)
+
+	// Run pruning pod on each node and pin it to that node
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		err = c.ensurePrunePod(prunerPodName, nodeStatus.NodeName, excludedIDs[len(excludedIDs)-1], excludedIDs, nodeStatus.TargetRevision)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func revisionKeyToStart(length, limit int) int {
+	if length > limit {
+		return length - limit
+	}
+	return 0
+}
+
+func (c *PruneController) ensurePrunePod(podName, nodeName string, maxEligibleRevision int, protectedRevisions []int, revision int32) error {
+	required := resourceread.ReadPodV1OrDie([]byte(prunerPod))
+	required.Name = fmt.Sprintf("%s-%d-%s", podName, revision, nodeName)
+	required.Namespace = c.targetNamespace
+	required.Spec.NodeName = nodeName
+	required.Spec.Containers[0].Image = c.prunerPodImageFn()
+	required.Spec.Containers[0].Command = c.command
+	required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args,
+		fmt.Sprintf("-v=%d", 4),
+		fmt.Sprintf("--max-eligible-id=%d", maxEligibleRevision),
+		fmt.Sprintf("--protected-ids=%s", revisionsToString(protectedRevisions)),
+		fmt.Sprintf("--resource-dir=%s", defaultResourceDir),
+		fmt.Sprintf("--static-pod-name=%s", c.podResourcePrefix),
+	)
+
+	if _, err := c.kubeClient.CoreV1().Pods(c.targetNamespace).Create(required); err != nil && !apierrors.IsAlreadyExists(err) {
+		c.eventRecorder.Eventf("PrunePodFailed", "Failed to create pod for %s: %v", resourceread.WritePodV1OrDie(required), err)
+		return err
+	}
+	return nil
+}
+
+func revisionsToString(revisions []int) string {
+	revisionString := ""
+	if len(revisions) > 0 {
+		for i := 0; i < len(revisions)-1; i++ {
+			revisionString = fmt.Sprintf("%s,%d", revisionString, revisions[i])
```

Comment:
this almost looks like you want to use `strings.Join` ? ;-)

Reply:
I do! but I have a slice of ints and `strings.Join` takes a slice of strings, so I figured this hack was about as many steps as converting the ints to strings which would still have to be one-by-one I think

Reply:
can probably come up with a better way to do this though